File: src/main/java/com/webapp/model/MenuEntry.java
package com.webapp.model;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.apache.ignite.cache.query.annotations.QuerySqlField;

import java.util.List;
import java.util.UUID;

@Getter
@Setter
@ToString
public class MenuEntry {

    @QuerySqlField(index = true)
    private String id;

    @QuerySqlField(index = true)
    private String parentId;

    @QuerySqlField(index = true)
    private String url;

    @QuerySqlField(index = true)
    private String name;

    private List<String> roles;
    
    public MenuEntry() {
        this.id = UUID.randomUUID().toString();
    }
}

File: addressbook-common/src/main/kotlin/com/addressbook/model/MenuEntry.kt
package com.addressbook.model

import org.apache.ignite.cache.query.annotations.QuerySqlField
import java.util.*
import javax.persistence.ElementCollection

@javax.persistence.Entity
@javax.persistence.Table(name = "menus")
@dev.morphia.annotations.Entity("menus")
class MenuEntry {

    @dev.morphia.annotations.Id
    @javax.persistence.Id
    @QuerySqlField(index = true)
    var id: String? = null

    @QuerySqlField(index = true)
    var parentId: String? = null

    @QuerySqlField(index = true)
    var url: String? = null

    @QuerySqlField(index = true)
    var name: String? = null

    @ElementCollection
    var roles: List<String>? = null

    init {
        this.id = UUID.randomUUID().toString()
    }
}
